Add a non breaking feature for module configs to support validation & rejection of config sets

KarthikSubbarao opened the pull request:
Add a non breaking feature for module configs to support validation & rejection of config sets

Also added integration testing for validation of new functionality and to ensure it is non breaking

Implements https://github.com/valkey-io/valkeymodule-rs/issues/123

dmitrypol commented:
nit:  could you please rebase with PR I just merged run `cargo fmt` please?

dmitrypol commented on examples/configuration.rs:
```diff
@@ -61,19 +115,23 @@ valkey_module! {
     configurations: [
         i64: [
             ["i64", &*CONFIGURATION_I64, 10, 0, 1000, ConfigurationFlags::DEFAULT, Some(Box::new(on_configuration_changed))],
+            ["reject-i64", &*CONFIGURATION_REJECT_I64, 10, 0, 1000, ConfigurationFlags::DEFAULT, Some(Box::new(on_configuration_changed)), Some(Box::new(on_i64_config_set::<ValkeyString, ValkeyGILGuard<i64>>))],
```
tiny nit:  could you stick to snake_case just to be consistent in all 4 examples

KarthikSubbarao commented:
Thank you for helping review this, @dmitrypol. I addressed the comments in this commit: https://github.com/valkey-io/valkeymodule-rs/pull/144/commits/809af261edabf345f1b236fab28956d2ee860058
